File: line_lengths/LineLengths.scala
import scala.io.Source

def widthOfLength(s: String) = s.length.toString.length
if (args.length > 0) {
  val lines = Source.fromFile(args(0)).getLines.toList
  val widths = lines.map(widthOfLength)
  val maxWidth = widths.reduceLeft((a,b) => a.max(b))
  for (line <- lines) {
    val padding = " " * (maxWidth - widthOfLength(line))
    print(padding + line.length + " " + line)
  }
}
else 
  Console.err.println("Please enter filename")

File: line_lengths/line_lengths.clj
(defn width-of-length [str]
  (.. str (length) (toString) (length)))
(defn read-lines [file]
  (re-seq #"[^\n]+" (slurp file)))
(defn padding [size]
  (apply str (replicate size " ")))
(defn print-lines [file]
  (let [lines (read-lines file)
 	      widths (map width-of-length lines)
	      max-width (reduce max widths)]
   (doseq line lines
     (println 
      (padding (- max-width (width-of-length line)))
      (. line (length)) 
      " " 
      line))))
